### PR TITLE
Fix unit test target, use +health option and add rudimentary tests

### PR DESCRIPTION
**Pull request by felixLam**

### CODE REVIEW

File: ISHPermissionKitTests/ISHPermissionKitTests.m [hunk 1]
```diff
@@ -7,33 +7,53 @@
...
 @interface ISHPermissionKitTests : XCTestCase
 
 @end
 
 @implementation ISHPermissionKitTests
 
-- (void)setUp {
-    [super setUp];
-    // Put setup code here. This method is called before the invocation of each test method in the class.
+- (void)testNonNilRequsts {
```
Comment (shagedorn) [inline ISHPermissionKitTests/ISHPermissionKitTests.m]: Requests

File: ISHPermissionKitTests/ISHPermissionKitTests.m [hunk 2]
```diff
@@ -7,33 +7,53 @@
...
-    // This is an example of a functional test case.
-    XCTAssert(YES, @"Pass");
-}
-
-- (void)testPerformanceExample {
-    // This is an example of a performance test case.
-    [self measureBlock:^{
-        // Put the code you want to measure the time of here.
-    }];
+- (void)testUnsupportedOnSimulator {
```
Comment (shagedorn) [inline ISHPermissionKitTests/ISHPermissionKitTests.m]: This test should be `#ifdef`'ed so it only runs in the simulator

**Comment by shagedorn** [comment]
What exactly was changed in the project/target setup?

File: ISHPermissionKitTests/ISHPermissionKitTests.m [hunk 3]
```diff
@@ -7,33 +7,53 @@
...
+                               @(ISHPermissionCategoryPhotoCamera),
+                               @(ISHPermissionCategoryNotificationLocal),
+                               @(ISHPermissionCategoryNotificationRemote),
+                               @(ISHPermissionCategorySocialFacebook),
+                               @(ISHPermissionCategorySocialTwitter),
+                               @(ISHPermissionCategorySocialSinaWeibo),
+                               @(ISHPermissionCategorySocialTencentWeibo),
+                               @(ISHPermissionCategoryAddressBook),
+                               @(ISHPermissionCategoryEvents),
+                               @(ISHPermissionCategoryReminders)
```
Comment (mollidor) [inline ISHPermissionKitTests/ISHPermissionKitTests.m]: let's add a comma here as well. It will help with the next diff.

**Comment by shagedorn** [comment]
*ping* @felixLam

**Comment by felixLam** [comment]
I have updated the travis config: it now builds all variants framework and static lib and runs the tests

**Comment by shagedorn** [comment]
👍

**Pull request merged**